src/kibana/components/vislib/visualizations/tile_map.js

```diff
@@ -208,36 +274,81 @@ define(function (require) {
      * @method shadedCircleMarkers
      * @param map {Object}
      * @param mapData {Object}
-     * @return {undefined}
+     * @return {Leaflet object} featureLayer
      */
     TileMap.prototype.shadedCircleMarkers = function (map, mapData) {
       var self = this;
 
-      // TODO: add UI to select local min max or all min max
+      // super min and max from all chart data
+      var min = mapData.properties.allmin;
+      var max = mapData.properties.allmax;
+
+      var featureLayer = L.geoJson(mapData, {
+        pointToLayer: function (feature, latlng) {
+          var count = feature.properties.count;
+          var radius = self.geohashMinDistance(feature);
+          return L.circle(latlng, radius);
+        },
+        onEachFeature: function (feature, layer) {
+          self.bindPopup(feature, layer);
+        },
+        style: function (feature) {
+          var count = feature.properties.count;
+          var color = self.quantizeColorScale(count, min, max);
+          return {
+            fillColor: color,
+            color: self.darkerColor(color),
+            weight: 1.0,
+            opacity: 1,
+            fillOpacity: 0.75
+          };
+        }
+      });
+
+      // add legend
+      if (mapData.features.length > 1) {
+        self.addLegend(mapData, map);
+      }
 
-      // min and max from chart data for this map
-      // var min = mapData.properties.min;
-      // var max = mapData.properties.max;
+      return featureLayer;
+    };
+
+    /**
+     * Type of data overlay for map:
+     * creates featurelayer from mapData (geoJson)
+     * with rectangles that show the geohash grid bounds
+     *
+     * @method geohashGrid
+     * @param map {Object}
+     * @param mapData {Object}
+     * @return {undefined}
+     */
+    TileMap.prototype.shadedGeohashGrid = function (map, mapData) {
+      var self = this;
 
       // super min and max from all chart data
       var min = mapData.properties.allmin;
       var max = mapData.properties.allmax;
 
-      var length = mapData.properties.length;
-      var precision = mapData.properties.precision;
-      var zoomScale = self.zoomScale(mapZoom);
       var bounds;
-      var defaultColor = '#005baa';
+
       var featureLayer = L.geoJson(mapData, {
         pointToLayer: function (feature, latlng) {
           var count = feature.properties.count;
-          var rad = zoomScale * 3;
-          return L.circleMarker(latlng, {
-            radius: rad
-          });
+          var gh = feature.properties.rectangle;
```

Comment:
`gh` is a pretty bad name, I have no idea what it's short for. Also, a note about what's going on here would be helpful, in particular, why [0,0], [0,1], [2,1], [2,0].